vizro-core/docs/pages/user-guides/custom-components.md

```diff
@@ -266,6 +266,224 @@ vm.Page.add_type("components", Jumbotron)
+        return is_open
+   ```
+
+Add the custom action `open_offcanvas` as a `function` argument inside the [`Action`][vizro.models.Action] model.
+
+
+??? example "Example of the use of custom component with actions"
+
+    === "app.py"
+        ``` py
+        from typing import List, Literal
+
+        import dash_bootstrap_components as dbc
+        import vizro.models as vm
+        import vizro.plotly.express as px
+        from dash import html
+        from vizro import Vizro
+
+        from vizro.models import Action
+        from vizro.models.types import capture
+
+
+        # 1. Create new custom component
+        class OffCanvas(vm.VizroBaseModel):
+            type: Literal["offcanvas"] = "offcanvas"
+            title: str
+            content: str
+
+            def build(self):
+                return html.Div(
+                    [
+                        dbc.Offcanvas(
+                            children=html.P(self.content),
+                            id=self.id,
+                            title=self.title,
+                            is_open=False,
+                        ),
+                    ]
+                )
+
+
+        # 2. Add new components to expected type - here the selector of the parent components
+        vm.Page.add_type("components", OffCanvas)
+
+        # 3. Create custom action
+        @capture("action")
+        def open_offcanvas(n_clicks, is_open):
+            if n_clicks:
+                return not is_open
+            return is_open
+
+        page = vm.Page(
+            title="Custom Component",
+            components=[
+                vm.Button(
+                    text="Open Offcanvas",
+                    id="open_button",
+                    actions=[
+                        vm.Action(
+                            function=open_offcanvas(),
+                            inputs=["open_button.n_clicks", "offcanvas.is_open"],
+                            outputs=["offcanvas.is_open"],
+                        )
+                    ],
+                ),
+                OffCanvas(
+                    id="offcanvas",
+                    content="OffCanvas content",
+                    title="Offcanvas Title",
+                ),
+            ],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+
+        ```
+    === "yaml"
+        ```yaml
+        # Custom components are currently only possible via python configuration
+        ```
+    === "Result"
+        [![CustomComponent3]][CustomComponent3]
+
+    [CustomComponent3]: ../../assets/user_guides/custom_components/customcomponent_3.gif
+
+
+### How to trigger actions with a custom component
+
+As mentioned above, custom components can trigger action. To enable the custom component to trigger the action, we need to add some additional lines of code:
+
+1. **Add the `actions` argument to your custom component**. The type of the `actions` argument is `List[Action]`.
```

Comment:
Do we need bolding here?

Reply:
I think so, it helps make it clear what the action (pardon the pun) in the sequence is.